src/main/java/com/configcat/ConfigurationProvider.java

```diff
@@ -55,13 +56,61 @@ public interface ConfigurationProvider extends Closeable {
      */
     <T> CompletableFuture<T> getValueAsync(Class<T> classOfT, String key, User user, T defaultValue);
 
+    /**
+     * Gets the value of a feature flag or setting as T identified by the given {@code key}.
+     *
+     * @param classOfT     the class of T. Only {@link String}, {@link Integer}, {@link Double} or {@link Boolean} types are supported.
+     * @param key          the identifier of the configuration value.
+     * @param defaultValue in case of any failure, this value will be returned.
+     * @param <T>          the type of the desired config value.
+     * @return the evaluation details.
+     */
+    <T> EvaluationDetails<T> getValueDetails(Class<T> classOfT, String key, T defaultValue);
+
+    /**
+     * Gets the value of a feature flag or setting as T identified by the given {@code key}.
+     *
+     * @param classOfT     the class of T. Only {@link String}, {@link Integer}, {@link Double} or {@link Boolean} types are supported.
+     * @param key          the identifier of the configuration value.
+     * @param user         the user object to identify the caller.
```

Comment:
```suggestion
     * @param user         the user object.
```
is it possible to add a link here?